src/app/@ansyn/core/reducers/core.reducer.ts

```diff
@@ -59,6 +65,10 @@ export function CoreReducer(state = coreInitialState, action: CoreActions | any)
 		case CoreActionTypes.SET_LAYOUT:
 			return {...state, layout: action.payload };
 
+		case CoreActionTypes.SET_WAS_WELCOME_NOTIFICATION_SHOWN_FLAG:
+			const payloadObj = {wasWelcomeNotificationShown: action.payload};
+			return {...state, ...payloadObj };
```

Comment:
use updateSession() here and remove onWelcomeNotification$ effect

Reply:
This should not be done, if I understand redux/ngrx. A reducer has to be a pure function. It should not call a side-effect function like updateSession(). This is what effects are for.